docs/install/authentication-mechanisms-gcp.md

```diff
@@ -138,7 +140,292 @@ gcloud projects add-iam-policy-binding $PROJECT_ID \
+     Type:     Ready
+     Status:   False
+     Message:  Failed to reconcile Pub/Sub topic: rpc error: code = Unauthenticated desc = transport: compute: Received 403 `
+               unable to generate token; IAM returned 403 Forbidden: The caller does not have permission
+               his error could be caused by a missing IAM policy binding on the target IAM service account.
+     Reason:   TopicReconcileFailed
+     ```
+       
+     ---
+   ***To solve this issue***, you can:
+   - Check the Google Cloud Service Account `cloud-run-events` for the Control Plane has all required permissions.
+   - Check authentication configuration is correct for the Control Plane. 
+     - If you are using Workload Identity for the Control Plane, 
+     refer [here](./authentication-mechanisms-gcp.md/#workload-identity) to check the
+     Google Cloud Service Account `cloud-run-events`, 
+     and the Kubernetes Service Account `controller` in namespace `cloud-run-events`.
+     - If you are using Kubernetes Secret for the Control Plane, 
+     refer [here](./authentication-mechanisms-gcp.md/#kubernetes-secrets) to check the 
+     Kubernetes Secret `google-cloud-key` in namespace `cloud-run-events`.
+     
+   ***Note:*** For Kubernetes Secret, if the JSON private key is no longer existing under your Google Cloud Service Account 
+   `cloud-run-events`. Then, even the Google Cloud Service Account `cloud-run-events` has all required permissions, and 
+   the corresponding Kubernetes Secret `google-cloud-key ` is in namespace `cloud-run-events`, you still get permission related error. 
+   To such case, you have to re-download the JSON private key and re-create the Kubernetes Secret, refer 
+   [here](./authentication-mechanisms-gcp.md/#option-2-kubernetes-secrets) for instructions.
+     
+1. ***Resources are READY, but can't receive Events***
+
+   Sometimes, a resource instance is READY, but it can't receive Events. It might be an authentication configuration problem, 
+   if the underlying `Deployment` doesn't have minimum availability.
+    
+   Typically, the name of an 
+   underlying `Deployment` for a resource instance could be a truncated version of 
+   `(prefix)-(resource-instance-name)-(uid)`. 
+   1. If the resource instance is a `Source`, the prefix is `cre-src`. 
+   1. If the resource instance is a `Channel`, the prefix is `cre-chan`. 
+   1. If the resource instance is a `Pullsubscription`, the prefix is `cre-ps`
+   
+   You can use the following command to check if the underlying `Deployment`'s available pod is zero.
+   
+   ```shell
+   kubectl get deployment -n resource-instance-namespace
+   ```
+    
+   ---  
+   Here is a detailed example checking a CloudAuditlogsSource `test`'s underlying `Deployment` in namespace `default`:
+     
+   After running the following command, you will find a `Deployment` named as 
+   `cre-src-cloudauditlogssource-te(UID)`.
+        
+   ```shell
+   kubectl get deployment test -n default
```

Comment:
Should this use the name `cre-src-cloudauditlogssource-te(UID)`, rather than `test`?

Reply:
Oh, It should use get all rather than a specific deployment, as this command is used for helping to find a deployment (but we don't know what exact name this deployment has). Thanks for pointing out this, I'll update it.